primitives/runtime/src/traits.rs

```diff
@@ -1164,7 +1166,11 @@ pub trait Header: Clone + Send + Sync + Codec + Eq + MaybeSerialize + Debug + 's
 		+ Copy
 		+ MaybeDisplay
 		+ AtLeast32BitUnsigned
-		+ Codec;
+		+ Default
+		+ TypeInfo
+		+ MaxEncodedLen
+		+ FullCodec
+		+ AsPrimitive<usize>;
```

Comment:
I find this bound to be a bit fishy.. why is it needed?

Reply:
As noted elsewhere, we should try and avoid bringing trait bounds here as much as possible. Things like `MEL` and `Codec` seem reasonable to me, and unavoidable. But, for example, if only one pallet requires `Number` to be `AsPrimitive<usize>`, this should not leak here. 

What we should do instead is to make that pallet define a custom block number associated type, which defines extra conditions. We do something similar with `type CurrencyBalance` in some pallets as well. A similar pattern might help here.

Reply:
Ideally, the trait bounds here should be things that _this crate_ thinks of block number, and FRAME should be able to express its won additional bounds. Not sure what is the right syntax to express this, but I think this is the right mental model to think about this.

Reply:
This was again needed for Cumulus which earlier defined a Substrate Chain directly as having a BlockNumber with this bound. Now, that the BlockNumber is retrieved from the Header, we cannot add that bound here, thus, having to leak it here. Please let me know if there's a better way.

Reply:
Where is this needed? Maybe Cumulus should  rather create a wrapper type.